executor/calibrate_resource.go

```diff
@@ -83,13 +91,72 @@ type baseResourceCost struct {
 	writeReqCount uint64
 }
 
+const (
+	valuableUsageThreshold = 0.2
+	lowUsageThreshold      = 0.1
+	percentOfPass          = 0.9
+	discardRate            = 0.1
+
+	maxDuration = time.Hour * 24
+	minDuration = time.Minute * 10
+)
+
 type calibrateResourceExec struct {
 	baseExecutor
-
+	optionList   []*ast.DynamicCalibrateResourceOption
 	workloadType ast.CalibrateResourceType
 	done         bool
 }
 
+func (e *calibrateResourceExec) checkDynamicCalibrateOptions() (startTime time.Time, endTime time.Time, err error) {
+	var dur time.Duration
+	var ts uint64
+	for _, op := range e.optionList {
+		switch op.Tp {
+		case ast.CalibrateStartTime:
+			ts, err = staleread.CalculateAsOfTsExpr(e.ctx, e.optionList[0].Ts)
```

Comment:
```suggestion
			ts, err = staleread.CalculateAsOfTsExpr(e.ctx, op.Ts)
```

Reply:
Seems the logic of this function is not well covered by unit test, please add some cases